sophiread/SophireadCLI/src/sophiread_core.cpp

```diff
@@ -166,6 +169,58 @@ void timedSaveEventsToHDF5(const std::string &out_events,
   spdlog::info("Save events to HDF5: {} s", elapsed / 1e6);
 }
 
+std::vector<std::vector<std::vector<unsigned int>>> initializeTOFImages(
+    double super_resolution, const std::vector<double> &tof_bin_edges) {
+  int dim_x = static_cast<int>(517 * super_resolution);
+  int dim_y = static_cast<int>(517 * super_resolution);
+  std::vector<std::vector<std::vector<unsigned int>>> tof_images(
+      tof_bin_edges.size() - 1);
+  for (auto &tof_image : tof_images) {
+    tof_image.resize(dim_y, std::vector<unsigned int>(dim_x, 0));
+  }
+  return tof_images;
+}
+
+void updateTOFImages(
+    std::vector<std::vector<std::vector<unsigned int>>> &tof_images,
+    const TPX3 &batch, double super_resolution,
+    const std::vector<double> &tof_bin_edges, const std::string &mode) {
+  int dim_x = static_cast<int>(517 * super_resolution);
+  int dim_y = static_cast<int>(517 * super_resolution);
+
+  const std::vector<const IPositionTOF *> entries = [&]() {
+    if (mode == "hit") {
```

Comment:
The "mode" thing might be better as a `enum class`

Reply:
we only have two, hits or neutron, and any invalid one is caught during arg parsing. I don't see a strong case of creating a enum class here.

Reply:
I feel like it the enum is easier to read. A constant representing the magic value would be equivalent.